src/Containers/Microsoft.NET.Build.Containers/Tasks/ComputeDotnetBaseImageAndTag.cs

```diff
@@ -113,16 +113,37 @@ public override bool Execute()
         }
         else
         {
-            var defaultRegistry = RegistryConstants.MicrosoftContainerRegistryDomain;
-            if (ComputeRepositoryAndTag(out var repository, out var tag))
+            if (TargetRuntimeIdentiriersAreValid())
             {
-                ComputedContainerBaseImage = $"{defaultRegistry}/{repository}:{tag}";
-                LogInferencePerformedTelemetry($"{defaultRegistry}/{repository}", tag!);
+                var defaultRegistry = RegistryConstants.MicrosoftContainerRegistryDomain;
+                if (ComputeRepositoryAndTag(out var repository, out var tag))
+                {
+                    ComputedContainerBaseImage = $"{defaultRegistry}/{repository}:{tag}";
+                    LogInferencePerformedTelemetry($"{defaultRegistry}/{repository}", tag!);
+                }
             }
             return !Log.HasLoggedErrors;
         }
     }
 
+    private bool TargetRuntimeIdentiriersAreValid()
+    {
+        var muslRidsCount = TargetRuntimeIdentifiers.Count(rid => rid.StartsWith("linux-musl", StringComparison.Ordinal));
```

Comment:
so we want either none or all to be linux-musl?
if ( TargetRuntimeIdentifiers[0].rid.StartsWith("linux-musl", StringComparison.Ordinal)
{
  return TargetRuntimeIdentifiers.All(rid => rid.StartsWith("linux-musl", StringComparison.Ordinal);
}
else
{
 return TargetRuntimeIdentifiers.All(rid => !rid.StartsWith("linux-musl", StringComparison.Ordinal);
}
note: I have no clue about the possible values of the muslRid or what the musl means.

Reply:
> so we want either none or all to be linux-musl? scratch my attempt to simplify, there is too much bookkeeping for it to work nicely.

yes, exactly. 

Basically, we need this validation, because for rids starting with "linux-musl" we choose alpine image as a base image. And because we compute base image once for all rids, we check that all are "linux-musl" to get alpine image.

Reply:
Actually, I think I can make this code shorter

Reply:
Nope, scratch that. making the code shorter will make it confusing

Reply:
> Basically, we need this validation, because for rids starting with "linux-musl" we choose alpine image as a base image. And because we compute base image once for all rids, we check that all are "linux-musl" to get alpine image.

This would be a great comment in that code ;-)